model/project_parser_test.go

```diff
@@ -1411,6 +1411,43 @@ buildvariants:
 			checkIsTaskGroupTaskUnit(t, bvtu)
 		}
 	})
+
+	t.Run("MaxHostsForTaskGroup", func(t *testing.T) {
+		validMaxHostYml := `
+tasks:
+- name: example_task_1
+- name: example_task_2
+task_groups:
+- name: example_task_group
+  max_hosts: -1
+  tasks:
+  - example_task_1
+  - example_task_2
+buildvariants:
+- name: bv
+  display_name: "bv_display"
+  tasks:
+  - name: example_task_group
+  - name: inline_task_group
+    task_group:
+      max_hosts: -1
+      tasks:
+      - example_task_1
+      - example_task_2
+`
+		proj := &Project{}
+		_, err := LoadProjectInto(ctx, []byte(validMaxHostYml), nil, "id", proj)
+		assert.NotNil(t, proj)
```

Comment:
From [the style guide](https://docs.google.com/document/d/1-DB1FQ_zb5fX_0pnVcwyU0HmJPrloFBu6qSE8dkk8PU/edit#heading=h.9946u3sq3e2n): `require.NotNil` instead of assert because if any logic that this test relies upon changes and makes the assertion wrong, `proj` will be nil, the later dereferences of `proj` fields will panic the entire Go runtime and stop all tests.

For example, this could happen if someone in the future changes the logic of `LoadProjectInto` or project translation that makes this error.